src/openms/include/OpenMS/ANALYSIS/ID/SiriusAdapterAlgorithm.h

```diff
@@ -54,19 +56,84 @@ namespace OpenMS
       SiriusAdapterAlgorithm();
 
       /// Struct for temporary folder structure
-      struct SiriusTmpStruct
+      struct SiriusTemporaryFileSystemObjects
       {
-        String tmp_dir;
-        String tmp_ms_file;
-        String tmp_out_dir;
-      };
+      public:
 
-      /**
-        @brief Construct temporary folder structure for SIRIUS (SiriusTmpStruct)
+        SiriusTemporaryFileSystemObjects(int debug_level) : removeable_(false) { constructSiriusTemporaryFileSystemObjects_(debug_level); }
+
+        ~SiriusTemporaryFileSystemObjects(){ if (removeable_) removeSiriusTemporaryFileSystemObjects_(debug_level_, tmp_dir_, tmp_ms_file_); }
+
+        String getTmpDir()
+        {
+          return tmp_dir_;
+        }
+
+        String getTmpOutDir()
+        {
+          return tmp_out_dir_;
+        } 
 
+        String getTmpMsFile()
+        {
+          return tmp_ms_file_;
+        }
+      
+      private:
+        bool removeable_;
+        int debug_level_;
+
+        String tmp_dir_;
+        String tmp_ms_file_;
+        String tmp_out_dir_;
+
+        /**
+        @brief Construct temporary folder structure for SIRIUS (SiriusTmpStruct)
+        @param debug_level Debug level
         @return SiriusTmpStruct
-      */
-      static SiriusAdapterAlgorithm::SiriusTmpStruct constructSiriusTmpStruct();
+        */
+
+        void constructSiriusTemporaryFileSystemObjects_(int debug_level)
+        {
+          QString base_dir = File::getTempDirectory().toQString();
+          tmp_dir_ = String(QDir(base_dir).filePath(File::getUniqueName().toQString()));
+          tmp_ms_file_ = QDir(base_dir).filePath((File::getUniqueName() + ".ms").toQString());
+          tmp_out_dir_ = QDir(tmp_dir_.toQString()).filePath("sirius_out");
+          removeable_ = true;
+          debug_level_ = debug_level;
+        } 
+
+        /**
+        @brief Removes temporary directory and file structure
+
+        @param debug_level Debug level
+        @param tmp_dir Path to temporary directory
+        @param tmp_ms_file Path to temporary ms file
+        */
+
+        void removeSiriusTemporaryFileSystemObjects_(int debug_level, const String& tmp_dir, const String& tmp_ms_file)
+        {
+          // clean tmp directory if debug level < 2 
```

Comment:
extra:
maybe add a 
`constexpr int debug_threshold = 2;`
and use it throughout the function to have a single place of reference for the required debug level (avoiding mentioning it multiple times which could go out of sync if we decide to change it at some point)